lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -2459,7 +2459,7 @@ namespace Js
             Var member;
 
             // If the item is found in the array own body, then it is a number
-            if (JavascriptOperators::GetOwnItem(object, index, &member, scriptContext))
+            if (JavascriptOperators::GetOwnItem(object, index, &member, scriptContext) && member != scriptContext->GetLibrary()->GetUndefined())
```

Comment:
@boingoing, will do.
@goyakin, I'd like to open a non-bug-non-1.2 issue to find out if it makes sense to change the behavior of GetItem(). In the meantime, I think it is prudent to avoid that in 1.2